File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        n = len(s)
        if n < 2:
            return s
        st = 0 
        en = 0
        def expand ( s , start , end ):
            while ( start >= 0 and end < len(s)) and s[start] == s[end]:
                start = start - 1
                end = end + 1

            return end - start - 1
        for i in range (n):
            ol = expand(s,i,i)
            el = expand(s,i,i+1)
            x = max(ol , el)

            if x > en - st + 1 :
                st = i - ( (x-1) // 2 )
                en = i +  ( x // 2 )
        return s[st : en + 1]

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        int n = s.length();
        int st = 0 ; 
        int e = 0 ;
        int len = Integer.MIN_VALUE;
        for(int i = 0 ; i < n ; i++){
            int ol = expand(s,i,i);
            int el = expand(s,i,i+1);
            len = Math.max(ol,el);
            if(len >  e - st + 1){
                st = i - (len - 1)/2;
                e = i + len / 2;
            }
        }
        return s.substring( st , e + 1 );
        
    }
    private int expand(String a  , int l , int r ){
        while( l >= 0 && r < a.length() && a.charAt(l) == a.charAt(r)  ){
            l--;
            r++;
        }
        return r - l - 1 ;
    }
}